grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/JavaGrpcClientStubFactory.java

```diff
@@ -51,7 +51,14 @@ public ServiceDescriptor findServiceDescriptor(Class<?> clientType) {
             final Method getServiceDescriptorMethod = enclosingClass.getDeclaredMethod("getServiceDescriptor");
             return (ServiceDescriptor) getServiceDescriptorMethod.invoke(null);
         } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
-            return null;
+            Throwable t = e;
+            if (e instanceof InvocationTargetException) {
+                final Throwable targetException = ((InvocationTargetException) e).getTargetException();
+                if (targetException != null) {
+                    t = targetException;
+                }
+            }
+            throw new ServiceDescriptorResolutionException(t);
```

Comment:
What do you think of adding the stub information in an error message like `KotlinGrpcClientStubFactory` to easily associate a stub factory with the exception?
The information will be useful when multiple exceptions are thrown.

Reply:
That is brilliant. 👍